pyopenms_viz/_plotly/core.py

```diff
@@ -793,26 +824,34 @@ def _add_box_boundaries(self, annotation_data, **kwargs):
                 legend_label = f"{use_name} (q-value: {feature['q_value']:.4f})"
             else:
                 legend_label = f"{use_name}"
+            # Plot rectangle
+            self.fig.add_shape(
+                type="rect",
+                x0=x0,
+                y0=y0,
+                x1=x1,
+                y1=y1,
+                line=dict(
+                    color=color,
+                    width=self.feature_config.line_width,
+                    dash=bokeh_line_dash_mapper(self.feature_config.line_type, "plotly"),
+                ),
+                fillcolor="rgba(0,0,0,0)",
+                opacity=0.5,
+                layer="above",
+            )
+            # Add a dummy trace for the legend
             self.fig.add_trace(
                 go.Scatter(
-                    x=[
-                        x0,
-                        x1,
-                        x1,
-                        x0,
-                        x0,
-                    ],  # Start and end at the same point to close the shape
-                    y=[y0, y0, y1, y1, y0],
+                    x=[None],
+                    y=[None],
                     mode="lines",
-                    fill="none",
-                    opacity=0.5,
                     line=dict(
                         color=color,
                         width=self.feature_config.line_width,
-                        dash=bokeh_line_dash_mapper(
-                            self.feature_config.line_type, "plotly"
-                        ),
+                        dash=bokeh_line_dash_mapper(self.feature_config.line_type, "plotly"),
                     ),
+                    showlegend=True,
```

Comment:
should legend showing be configurable?

Reply:
I suppose we could make it optional for 3D plots as we have the in-plot annotation labels with this PR. For 2D-plots, the legend is currently the only way to pick a location so I am not sure how useful that would be.

We could add the same functionality but I am unsure how to best choose a good location for the in-plot annotation labels. Maybe we could allow for additional (x, y)-label-locations  in `annotation_data` and let the user decide? What do you think?

Reply:
Is this for adding in-plot text annotations 2D plots, that requires creating a dummy legend?

Reply:
Any updates on this? I am not too familiar with Plotly so not sure what is best

Reply:
> Is this for adding in-plot text annotations 2D plots, that requires creating a dummy legend?

A dummy legend is not necessary for in-plot annotations afaik. I was just wondering how useful making the legend optional would be if there is no alternative. Thus, I suggested adding annotation labels to 2D-Plots as well but am unsure about a good general placement for the label.

Reply:
I suppose we could require a set of coordinates for the user. This could be an optional feature for the 3D-Plots as well.